### PR TITLE
Add Noise Figure Circles

### PR DESCRIPTION
**Pull request by ericwrice**
Following the gain and stability circle methods, this pull request adds noise figure circle capability to networks. Verified against Pozar example 12.5 and the tests run with a few points from the existing ntwk_noise.s2p verified in Microwave Office. 

I am also curious to hear how other users prefer to interact with these circle features. Personally, I would like methods that return the center and radius of the circles and separate methods to plot the circles directly. I rarely use the loci of the circles for purposes other than plotting.

### CODE REVIEW

**Review by jhillairet — APPROVED** [review]
LGTM. 

Are you OK with the [rendered equations](https://scikit-rf--1082.org.readthedocs.build/en/1082/api/generated/skrf.network.Network.nf_circle.html#skrf.network.Network.nf_circle) ? 

As I don't use stability circle myself, I let other to give their advices.

**Comment by ericwrice** [comment]
Equations are good, made a quick change to make the alignment nicer.

File: skrf/network.py [hunk 1]
```diff
@@ -4515,6 +4517,105 @@ def gain_circle(self, target_port: int, gain: float, npoints: int = 181) -> np.n
...
+        nf : float
+            Noise figure of network in decibels.
+        npoints : int, optional
+            The number of points on the circumference of the circle.
+            More points result in a smoother circle, but require more computation. Default is 181.
+
+        Returns
+        -------
+        nfc : :class:`numpy.ndarray` (shape is `npoints x f`)
+            Loci of gain circles in complex numbers
```
Comment (keikawa) [inline skrf/network.py:4551]: Typo: noise figure circles

File: skrf/network.py [hunk 2]
```diff
@@ -4515,6 +4517,105 @@ def gain_circle(self, target_port: int, gain: float, npoints: int = 181) -> np.n
...
+        Example
+        --------
+        >>> import skrf as rf
+        >>> import matplotlib.pyplot as plt
+
+        Create a two-port network object
+
+        >>> ntwk = rf.Network('fet.s2p')
+
+        Calculate the source gain circles for all the frequencies at a noise figure of 1 dB
```
Comment (keikawa) [inline skrf/network.py:4562]: Typo: noise figure circles

Comment (ericwrice) [reply skrf/network.py]: Good catch, thank you! Fixed.

**Review by jhillairet — APPROVED** [review]

**Comment by jhillairet** [comment]
@ericwrice eventually, adding some documentation in the tutorial section about gain & stability circle & noise figure would be a nice addition!

**Comment by ericwrice** [comment]
@jhillairet, I agree, I'd be happy to add that!

**Pull request merged**